bin/picca_deltas.py

```diff
@@ -434,44 +488,46 @@ def main():
                                          bi_max=args.bi_max,
-                                         order=args.order,
                                          best_obs=args.best_obs,
                                          single_exp=args.single_exp,
                                          pk1d=args.delta_format,
                                          spall=args.spall)
 
+    #-- Add order info
+    for pix in data:
+        for forest in data[pix]:
+            if not forest is None:
+                forest.order = args.order 
+
     ### Read masks
     mask_obs_frame = None
     mask_rest_frame = None
     mask_rest_frame_dla = None
     if args.mask_file is not None:
         args.mask_file = os.path.expandvars(args.mask_file)
         try:
-            mask_obs_frame = []
-            mask_rest_frame = []
-            mask_rest_frame_dla = []
-            with open(args.mask_file, 'r') as file:
-                loop = True
-                for line in file:
-                    if line[0] == '#':
-                        continue
-                    cols = line.split()
-                    if cols[3] == 'OBS':
-                        mask_obs_frame += [[float(cols[1]), float(cols[2])]]
-                    elif cols[3] == 'RF':
-                        mask_rest_frame += [[float(cols[1]), float(cols[2])]]
-                    elif cols[3] == 'RF_DLA':
-                        mask_rest_frame_dla += [[
-                            float(cols[1]), float(cols[2])
-                        ]]
-                    else:
-                        raise ValueError("Invalid value found in mask")
-            mask_obs_frame = np.log10(np.asarray(mask_obs_frame))
-            mask_rest_frame = np.log10(np.asarray(mask_rest_frame))
-            mask_rest_frame_dla = np.log10(np.asarray(mask_rest_frame_dla))
-            if mask_rest_frame_dla.size == 0:
+            mask = Table.read(
+                args.mask_file, 
+                names=('type', 'wave_min', 'wave_max', 'frame'), 
+                format='ascii')
+            if 'OBS' in mask['frame']:
+                w = mask['frame'] == 'OBS'
+                mask_obs_frame = np.array([np.log10(mask['wave_min'])[w], 
+                                           np.log10(mask['wave_max'])[w]]).T
+            else:
+                mask_obs_frame = np.array([])
+            if 'RF' in mask['frame']:
+                w = mask['frame'] == 'RF'
+                mask_rest_frame = np.array([np.log10(mask['wave_min'])[w], 
+                                            np.log10(mask['wave_max'])[w]]).T
+            else:
+                mask_rest_frame = np.array([])
+            if 'RF_DLA' in mask['frame']:
+                w = mask['frame'] == 'RF_DLA'
+                mask_rest_frame_dla = np.array([np.log10(mask['wave_min'])[w], 
+                                                np.log10(mask['wave_max'])[w]]).T
+            else:
```

Comment:
Yes. It's still a lot of noisy code. I was planning to send a Table directly to forest.mask instead of these arrays.

Reply:
yes, that would actually be a good idea to keep everything more clean. Generating one table of those and then passing them to mask and add_dla and getting the right columns from there. This would also be more flexible in case one would like to later add additional stuff to the masking, e.g. BAL_RF or something...

Reply:
You seem to agree on this (and I do as well), but are we thinking to add this in part 2?

Reply:
Couldn't one even add it in here? Sounds sufficiently simple, I'll have a look how much of a change this would be

Reply:
Ok, did the changes, it affects `data.py`, `picca_deltas.py` and `picca_plotSpec.py`. Should I commit it in here, or should we just leave this as is and I open a PR?

Reply:
I think you can commit it here. Thanks!

Reply:
Done!